### PR TITLE
fix: update color tokens for modal component

### PR DESCRIPTION
**Pull request by tulasigudibanda**
Updated to theme based color tokens.

## ADO Story or GitHub Issue Link

[TASK 2003584](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003584)

## Figma Link

https://www.figma.com/design/CQuDZEeLiuGiALvCWjAKlu/branch/qMpff4GuFUEcsMUkvacS3U/Applications---Component-Library?node-id=11285-770&node-type=instance&m=dev

## Notes

Updated tokens in below components :
Modal

### CODE REVIEW

**Review by kumarivijeta — APPROVED** [review]

File: src/components/reusable/modal/modal.scss [hunk 1]
```diff
@@ -169,4 +172,5 @@ h1 {
   overflow-y: auto;
   margin: -4px;
   padding: 4px 16px 4px 4px;
+  background: var(--kd-color-background-accent-subtle);
```
Comment (brian-patrick-3) [inline src/components/reusable/modal/modal.scss:175]: I don't think we actually want a background color on the body here. I think that was just done in Figma to indicate a slot.

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]

**Comment by brian-patrick-3** [comment]
I think the designers also added some styles for the X button states?

**Comment by tulasigudibanda** [comment]
> I think the designers also added some styles for the X button states?

@brian-patrick-3  , included these styles.

File: src/components/reusable/modal/modal.scss (hunk 1 above)
Comment (tulasigudibanda) [reply src/components/reusable/modal/modal.scss:175]: Sure @brian-patrick-3 . Will check with Chandan

**Comment by brian-patrick-3** [comment]
> @brian-patrick-3 , included these styles.

Hmm, they don't seem to match Figma at all to me

**Comment by brian-patrick-3** [comment]
> Yes, they look different , may be because it blends with the modal background color . But I used same tokens as in figma
I think the colors are correct, but the size and shape of the button, and the outline/border thickness on active/focus, seems wrong.

File: src/components/reusable/modal/modal.scss [hunk 2]
```diff
@@ -10,9 +10,10 @@ dialog {
   @include elevation.shadow(3);
   padding: 32px;
   color: inherit;
-  background: var(--kd-color-background-ui-default);
+  background: var(--kd-color-background-container-default);
   border-radius: 8px;
-  border: 1px solid var(--kd-color-border-light);
+  border: 1px solid var(--kd-color-border-variants-light);
+  box-shadow: 0px 2px 12px 0px rgba(61, 60, 60, 0.25);
```
Comment (brian-patrick-3) [inline src/components/reusable/modal/modal.scss:16]: This box-shadow shouldn't be needed, as Elevation is already included.

**Comment by tulasigudibanda** [comment]
> > Yes, they look different , may be because it blends with the modal background color . But I used same tokens as in figma
> 
> I think the colors are correct, but the size and shape of the button, and the outline/border thickness on active/focus, seems wrong.

Pushed the changes

File: src/components/reusable/modal/modal.scss (hunk 1 above)
Comment (tulasigudibanda) [reply src/components/reusable/modal/modal.scss]: Confirmed with UX. Removed the background color on the body

File: src/components/reusable/modal/modal.scss [hunk 3]
```diff
@@ -88,20 +89,22 @@ button {
...
 
   &:hover {
-    color: var(--kd-color-text-link-hover);
+    background: var(--kd-color-background-ui-hollow-hover);
   }
 
   &:active {
-    color: var(--kd-color-text-pressed);
+    border-radius: 2px;
+    border: 1px solid var(--kd-color-border-ui-pressed);
```
[the remark this answers is not in the document]
Comment (tulasigudibanda) [reply src/components/reusable/modal/modal.scss:105]: Added 1px transparent border . Thanks Brian

**Review by brian-patrick-3 — APPROVED** [review]

**Comment by tulasigudibanda** [comment]
> Hi @tulasigudibanda, Hide cancel is not hiding the cancel button in Modal and before close component, can you please check.! image

@srpriyankashetty ,
Hide cancel is a storybook doc update ,not related to colors. Pushed the change anyway.
Rebuilt and button show colors as per the latest tokens. 
Screenshot 2024-12-12 at 2 42 33 PM

**Comment by srpriyankashetty** [comment]
Verified

**Review by srpriyankashetty — APPROVED** [review]
Verified Modal component in Light and dark mode theme and looks fine
Verified the OK button actions and color changes in Modal, Action button, before close and with form and looks fine
Verified the hide close and looks fine
Verified the Secondary button color in both dark and light mode theme

**Pull request merged**